### PR TITLE
docs: describe our telemetry

### PR DESCRIPTION
**Pull request by ammario**
This PR proposes features that do not yet exist. I figured we could use this PR as both production documentation and an RFC.

### CODE REVIEW

**Review by tjcran — COMMENTED** [review]
@ammario I think this all makes sense and I agree with/like the approach. Other OSS has set the precedent of not having what you describe as `level 0` ie you can't disable telemetry; but you can set it to an anonymized setting aka `level 1`. So I think the decision we have is whether we want to include `level 0` in Enterprise or forego `level 0` altogether. 

I think having the ability to opt out entirely (even if it is only for Enterprise customers) is a good move. But I'm curious to see what others think.

**Comment by ammario** [comment]
@tjcran thanks for the review. For our customer base level 0 will be necessary. Perhaps it won't be default in the enterprise plan, but it will certainly be requested often.

**Review by tjcran — APPROVED** [review]

**Comment by ammario** [comment]
@tjcran someone will need someone to build out the feature before I can merge this. I think that should be really easy though, as we just need to add email addresses and then the environment variable parsing.

**Review by bpmct — APPROVED** [review]

**Comment by mafredri** [comment]
Have we considered the implications of `CODER_TELEMETRY=2` being the default for European users? I'm not an expert, but I believe it would be problematic under GDPR.

From [gdpr.eu/gdpr-consent-requirements](https://gdpr.eu/gdpr-consent-requirements/):
> Recital 43 discusses [freely given consent](https://gdpr.eu/Recital-43-Freely-given-consent/). It explains that you must get separate consent for each data processing operation. So if you want their [email address](https://proton.me/business?ref=gdpreu) for marketing purposes and their IP address for website analytics purposes, you must give the user an opportunity to confirm or decline each use.

**Review by f0ssel — APPROVED** [review]
I think options 1 and 2 should have some sort of mention of the data we do collect. So like "this level collects product usage patterns, email and IP. This one only collect product usage patterns. This one collects nothing. " We can be vague here, but I was confused on the difference between 1 and 0 because it reads like we *only* collect email and IP.

**Comment by f0ssel** [comment]
Love the overall message and rationale here ❤️

File: docs/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+First of all, we do not collect any information that could threaten the security of your
+installation. For example, we do not collect parameters,
+environment variables, or passwords.
+
+You can find a full list of the data we collect in the source code [here](https://github.com/coder/coder/blob/main/coderd/telemetry/telemetry.go).
+
+We offer three levels of telemetry that can be configured with the
+`CODER_TELEMETRY=x` environment variable. For example, telemetry level 1 can
+be configured with `CODER_TELEMETRY=1`.
```
Comment (kylecarbs) [inline docs/telemetry.md:15]: Let's adjust these docs to describe the current behavior, and make an issue if we want to change it.

Right now `CODER_TELEMETRY` can be set to `true` or `false`, and defaults to `true`.

**Review by kylecarbs — DISMISSED** [review]

File: docs/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+deployment admin, but will never use these emails for outbound marketing.
+
+`CODER_TELEMETRY=1` is our lightweight telemetry setting. It excludes
+email and IP addresses, but everything else in the aforementioned source
+code is still sent out. In this level, it is nearly impossible for Coder
+to associate an installation to specific organizations or users.
+
+`CODER_TELEMETRY=0` disables telemetry altogether. We reserve this setting
+for our enterprise customers. You can also reach out to contact@coder.com if
+you need a zero-telemetry license due to security policy requirements.
```
Comment (bpmct) [inline docs/telemetry.md:29]: Based on the current behavior

```suggestion
Telemetry that can be configured with the `CODER_TELEMETRY=x` environment variable.
For example, telemetry can be disabled with `CODER_TELEMETRY=false`.

`CODER_TELEMETRY=true` is our default level. It includes user email and
IP addresses. This information is used in aggregate to understand where our
users are and general demographic information. We may reach out to the
deployment admin, but will never use these emails for outbound marketing.

`CODER_TELEMETRY=false` disables telemetry altogether.
```

**Comment by ammario** [comment]
Well, weren't we making it required? I think of this doc as kind of a commitment to the community so would rather merge it after we have these uncertainties worked out.

File: docs/manifest.json [hunk 3]
```diff
@@ -134,6 +134,18 @@
       "icon_path": "./images/icons/users.svg",
       "path": "./users.md"
     },
+    {
+      "title": "Admin",
+      "description": "How to install and deploy Coder",
+      "icon_path": "./images/icons/cog.svg",
```
Comment (ghuntley) [inline docs/manifest.json:140]: 🚧 `"icon_path": "./images/icons/cog.svg",` 

I'm not sure (yet) where `/images/icons` is being imported from and picked `cog.svg` to match the pattern from Coder v1 documentation. Will need changing as not sure if `cog.svg` is even a thing...

CleanShot 2022-09-07 at 12 33 55@2x

File: docs/manifest.json [hunk 4]
```diff
@@ -134,6 +134,18 @@
       "icon_path": "./images/icons/users.svg",
       "path": "./users.md"
     },
+    {
+      "title": "Admin",
+      "description": "How to administer and manage Coder",
+      "icon_path": "./images/icons/cog.svg",
```
Comment (ghuntley) [inline docs/manifest.json:140]: 🚧 `"icon_path": "./images/icons/cog.svg",` 

I'm not sure (yet) where `/images/icons` is being imported from and picked `cog.svg` to match the pattern from Coder v1 documentation. Will need changing as not sure if `cog.svg` is even a thing...

CleanShot 2022-09-07 at 12 33 55@2x

File: docs/telemetry.md (hunk 1 above)
Comment (ghuntley) [reply docs/telemetry.md]: > Let's adjust these docs to describe the current behavior, and make an issue if we want to change it.

👍

File: docs/manifest.json (hunk 4 above)
Comment (ammario) [reply docs/manifest.json]: You can see how the docs lay out by following this workflow: https://github.com/coder/coder.com#docs-development.

**Pull request merged**